pkg/standalone/bundle.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package standalone
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+)
+
+const bundleDetailsFileName = "details.json"
+
+type bundleDetails struct {
+	RuntimeVersion    *string `json:"daprd"`
+	DashboardVersion  *string `json:"dashboard"`
+	CLIVersion        *string `json:"cli"`
+	BinarySubDir      *string `json:"daprBinarySubDir"`
+	ImageSubDir       *string `json:"dockerImageSubDir"`
+	DaprImageName     *string `json:"daprImageName"`
+	DaprImageFileName *string `json:"daprImageFileName"`
+}
+
+func (b *bundleDetails) parseDetails(detailsFilePath string) error {
+	bytes, err := ioutil.ReadFile(detailsFilePath)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(bytes, &b)
+	if err != nil {
+		return err
+	}
+	if b.RuntimeVersion == nil || b.DashboardVersion == nil || b.DaprImageName == nil || b.DaprImageFileName == nil {
```

Comment:
How does this behave if daprd is present in json but with empty value?

Reply:
good catch ... fixing the check ...